vowpalwabbit/loss_functions.cc

```diff
@@ -358,6 +358,58 @@ class poisson_loss : public loss_function
   }
 };
 
+class modified_hubber_loss : public loss_function
+{
+ public:
+  std::string getType() { return "modified_hubber"; }
+
+  float getLoss(shared_data*, float prediction, float label)
+  {
+    if (label != -1.f && label != 1.f)
+      std::cout << "You are using label " << label << " not -1 or 1 as loss function expects!" << std::endl;
+     float e = max(0, 1-prediction*label);
+    return (prediction*label >= -1.f) ? e*e : -4*label*prediction;
+  }
+
+  float getUpdate(float prediction, float label, float update_scale, float pred_per_update)
+  {
+
+  }
+
```

Comment:
I don't understand how updates work so anyone can explain then I will update it accordingly.

Reply:
i believe the update functions represent how to update the parameters.
e.g. squared loss is (label - pred)^2, and the parameter (weight) update is 2*(label-pred)

@sharatsc can you comment on the other functions, particularly the safe update?

Reply:
@gramhagen , @sharatsc 
I am implementing this loss so that we can use it for `SGD` in core VW. As [sklearn `SGD`](https://github.com/scikit-learn/scikit-learn/blob/95d4f0841d57e8b5f6b2a570312e9d832e69debc/sklearn/linear_model/_stochastic_gradient.py#L1071)  has this loss also so I think we can implement it for the VW.